sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/MetricReaderFactoryTest.java

```diff
@@ -92,21 +99,28 @@ void create_PeriodicConfigured() {
             .build();
     cleanup.addCloseable(expectedReader);
 
-    io.opentelemetry.sdk.metrics.export.MetricReader reader =
+    MetricReaderAndCardinalityLimits readerAndCardinalityLimits =
         MetricReaderFactory.getInstance()
             .create(
                 new MetricReaderModel()
                     .withPeriodic(
                         new PeriodicMetricReaderModel()
                             .withExporter(
-                                new PushMetricExporterModel().withOtlp(new OtlpMetricModel()))
-                            .withInterval(1)),
+                                new PushMetricExporterModel()
+                                    .withOtlpHttp(new OtlpHttpMetricExporterModel()))
+                            .withInterval(1)
+                            .withCardinalityLimits(new CardinalityLimitsModel().withDefault(100))),
                 spiHelper,
                 closeables);
+    MetricReader reader = readerAndCardinalityLimits.getMetricReader();
     cleanup.addCloseable(reader);
     cleanup.addCloseables(closeables);
 
     assertThat(reader.toString()).isEqualTo(expectedReader.toString());
+    assertThat(
+            Objects.requireNonNull(readerAndCardinalityLimits.getCardinalityLimitsSelector())
```

Comment:
the test will still happily fail without the `requireNonNull` check...I say just let it NPE. It'll be obvious.